crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -26,6 +32,134 @@ struct Metrics {
     native_price_cache_outdated_entries: IntGauge,
 }
 
+/// Trait for fetching a batch of native price estimates.
+#[mockall::automock]
+#[async_trait]
+pub trait NativePriceBatchFetcher: Sync + Send {
```

Comment:
This trait will be implemented by any provider which offer batching requests. In the upcoming PR it will be implemented by `CoinGecko`.